h2o-core/src/main/java/hex/PDP.java

```diff
@@ -0,0 +1,178 @@
+package hex;
+
+import jsr166y.CountedCompleter;
+import water.*;
+import water.api.schemas3.KeyV3;
+import water.api.schemas3.PDPV3;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.util.Log;
+import water.util.TwoDimTable;
+
+import java.util.Arrays;
+
+/**
+ * Create a Frame from scratch
+ * If randomize = true, then the frame is filled with Random values.
+ */
+public class PDP extends Lockable<PDP> {
+  transient final public Job _job;
+  public Key<Model> _model_id;
+  public Key<Frame> _frame_id;
+  public String[] _cols;
+  public int _nbins = 20;
+  public TwoDimTable[] _partial_dependence_data; //OUTPUT
+
+  public PDP(Key<PDP> dest) {
+    super(dest);
+    _job = new Job<>(dest, PDP.class.getName(), "PDP");
+  }
+
+  public Job<PDP> execImpl() {
+    sanityChecks();
+    delete_and_lock(_job);
+    _frame_id.get().read_lock(_job._key);
+    _model_id.get().read_lock(_job._key);
+    _job.start(new PDPDriver(), _cols.length);
+    return _job;
+  }
+
+  private void sanityChecks() {
+    if (_cols==null) {
+      Model m = _model_id.get();
+      if (m==null) throw new IllegalArgumentException("Model not found.");
+      Frame f = _frame_id.get();
+      if (f==null) throw new IllegalArgumentException("Frame not found.");
+
+      if (Model.GetMostImportantFeatures.class.isAssignableFrom(m.getClass())) {
+        _cols = ((Model.GetMostImportantFeatures)m).getMostImportantFeatures(10);
```

Comment:
This is not a sanity check: you are filling in the `_cols` parameter. Should be either moved out to a separate method, or this method renamed into `sanityChecksAndParameterFilling` to avoid confusion.